src/main/js/MergeModal.tsx

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import {Button, Modal, SubmitButton} from "@scm-manager/ui-components";
+import {WithTranslation, withTranslation} from "react-i18next";
+import MergeForm from "./MergeForm";
+import {MergeCommit, PullRequest} from "./types/PullRequest";
+import {Link} from "@scm-manager/ui-types";
+
+type Props = WithTranslation & {
+  close: () => void;
+  merge: (strategy: string, mergeCommit: MergeCommit) => void;
+  pullRequest: PullRequest;
+};
+
+type State = {
+  mergeStrategy: string;
+  mergeCommit: MergeCommit;
+  loading: boolean;
+};
+
+class MergeModal extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      mergeStrategy: "MERGE_COMMIT",
+      mergeCommit: {
+        commitMessage: "",
+        source: this.props.pullRequest.source,
+        target: this.props.pullRequest.target,
+        author: this.props.pullRequest.author
+      },
+      loading: false
+    };
+  }
+
+  selectStrategy = (strategy: string) => {
+    this.setState({
+      mergeStrategy: strategy
+    });
+  };
+
+  onChangeCommitMessage = (newMessage: string) => {
+    this.setState({ mergeCommit: { ...this.state.mergeCommit, commitMessage: newMessage } });
+  };
+
+  performMerge = () => {
+    const { merge } = this.props;
+    const { mergeStrategy, mergeCommit } = this.state;
+
+    this.setState({ loading: true });
+    merge(mergeStrategy, mergeCommit);
+  };
+
+  shouldCommitMessageExists = () => {
+    const { mergeStrategy, mergeCommit } = this.state;
+    return mergeStrategy != "FAST_FORWARD_IF_POSSIBLE" && mergeCommit.commitMessage == "";
```

Comment:
Here we should check against `commitMessage.trim()`, because a commit message consisting only of spaces is invalid, too

Reply:
you are right, thanks